designer/server/src/lib/forms.test.js

```diff
@@ -0,0 +1,521 @@
+      jest.mocked(forms.get).mockResolvedValueOnce({
+        ...formMetadata,
+        // Switch draft with live for test
+        live: formMetadata.draft,
+        draft: undefined
+      })
+
+      const options = {
+        method: 'get',
+        url: '/library/my-form-slug',
+        auth
+      }
+
+      await renderResponse(server, options)
+
+      const $card = document.querySelector('.app-form-card')
+      const $buttons = $card?.querySelectorAll('.govuk-button')
+
+      expect($buttons).toHaveLength(1)
+      expect($buttons?.[0]).toHaveTextContent('Create draft to edit')
+    })
+
+    it('Form overview has live buttons in side bar', async () => {
+      jest.mocked(forms.get).mockResolvedValueOnce(formMetadata)
+
+      const options = {
+        method: 'get',
+        url: '/library/my-form-slug',
+        auth
+      }
+
+      await renderResponse(server, options)
+
+      const $card = document.querySelector('.app-form-card')
+      const $buttons = $card?.querySelectorAll('.govuk-button')
+
+      expect($buttons).toHaveLength(2)
+      expect($buttons?.[0]).toHaveTextContent('Edit draft')
+      expect($buttons?.[1]).toHaveTextContent('Make draft live')
+    })
+  })
+
+  describe('Forms functions', () => {
+    describe('list', () => {
+      const token = auth.credentials.token
+
+      beforeEach(() => {
+        jest.clearAllMocks()
+      })
+
+      it('should fetch list of forms without pagination', async () => {
+        const mockResponse = {
+          data: [formMetadata],
+          meta: {}
+        }
+
+        jest.spyOn(fetch, 'getJson').mockResolvedValueOnce({
+          /** @type {any} */
+          response: {},
+          body: mockResponse
+        })
+
+        const result = await forms.list(token)
+
+        expect(fetch.getJson).toHaveBeenCalledWith(expect.any(URL), {
+          headers: { Authorization: `Bearer ${token}` }
+        })
+        expect(result).toEqual(mockResponse)
+      })
+
+      it('should fetch list of forms with pagination', async () => {
+        const options = { page: 2, perPage: 10 }
+        const mockResponse = {
+          data: [formMetadata],
+          meta: {
+            pagination: {
+              page: 2,
+              perPage: 10,
+              totalPages: 3,
+              totalItems: 25
+            }
+          }
+        }
+
+        jest.spyOn(fetch, 'getJson').mockResolvedValueOnce({
+          /** @type {any} */
+          response: {},
+          body: mockResponse
+        })
+
+        const result = await forms.list(token, options)
+
+        expect(fetch.getJson).toHaveBeenCalledWith(expect.any(URL), {
+          headers: { Authorization: `Bearer ${token}` }
+        })
+
+        const fetchGetJsonMock = /** @type {jest.Mock} */ (fetch.getJson)
+
+        /** @type {Array<[URL, object]>} */
+        const mockCalls = fetchGetJsonMock.mock.calls
+
+        const calledUrl = /** @type {URL} */ (mockCalls[0][0])
```

Comment:
Not ideal, but due to type checking issues when accessing mock.calls on the mocked fetch.getJson func